### PR TITLE
Rationalise header inlined assets

### PR DESCRIPTION
**Pull request by sndrs**
I keep finding myself bewildered when moving around the header files.

This moves as much of the header JS and CSS from of `javaScriptFirstSteps.scala.html` and `javaScriptLaterSteps.scala.html` to static includes as possible. 

This has the added bonus of ensuring most of the header JS now gets linted and uglified.

it _ought_ to be fine, but flagged for testing as it potentially quite breaking.
# To test
- [x] anything broken cross browser. if it causes an error it's likely to be quite obvious...

cc/ @rich-nguyen – would you mind sanity checking this?

### CODE REVIEW

**Comment by rich-nguyen** [comment]
i think this is possible to do with scala.js files like `serviceworker.scala.js`. This keeps the javascript within the template system when possible.

**Comment by rich-nguyen** [comment]
...plus it allows you to render config

**Comment by sndrs** [comment]
ooh nice, wasn't aware of that, cheers

**Comment by sndrs** [comment]
@rich-nguyen so this now uses `scala.js` templates. had zero joy with sbt uglify so far, but given we don't currently do that anyway, will raise that as a separate PR.

**Comment by michaelwmcnamara** [comment]
lazy-loading on network front broken, also the containers tha have loaded look unfinished - like flex hasn't run successfully on them.
console shows a syntax error:
Uncaught SyntaxError: Unexpected token ILLEGAL
for the /uk stykesheet - line 1068.
Looking at this in debugger shows:     "\n\n\n"

**Comment by sndrs** [comment]
@michaelwmcnamara should be sorted now, cheers

**Comment by michaelwmcnamara** [comment]
+1

**Comment by michaelwmcnamara** [comment]
Oops - wrong button :-E

File: grunt-configs/uglify.js [hunk 1]
```diff
@@ -16,8 +16,12 @@ module.exports = function(grunt, options) {
             ]
         },
         conf: {
-            files:[
-            {
+            files: [{
+                expand: true,
+                cwd: 'static/src/javascripts/header-inline',
```
Comment (OliverJAsh) [inline grunt-configs/uglify.js]: Is this used?

**Comment by OliverJAsh** [comment]
Only just seen this. I need to get better at reviewing PRs…

So much this :+1: Nice work!

File: grunt-configs/uglify.js (hunk 1 above)
Comment (sndrs) [inline grunt-configs/uglify.js]: no you're right, left over. i'll get rid of it after merging

**Pull request merged**